### PR TITLE
Update terminal to use xterm.js

### PR DESCRIPTION
**Pull request by jwforres**

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
     "selectize": "0.12.1",
     "c3": "0.4.11",
     "kubernetes-label-selector": "1.4.4",
-    "kubernetes-container-terminal": "1.0.4",
+    "kubernetes-container-terminal": "git@github.com:jwforres/container-terminal.git#xterm",
```
Comment (jwforres) [inline bower.json]: Needs to switch to the release once the upstream PR https://github.com/kubernetes-ui/container-terminal/pull/25 is merged

**Comment by jwforres** [comment]
@spadgett went ahead and opened this since it contains two less fixes that need to happen in the console repo

File: app/styles/_container-terminal.less [hunk 2]
```diff
@@ -3,6 +3,9 @@
   background: #000;
   height: 100%;
   width: 100%;
+  &.disconnected .fullscreen-toggle {
```
Comment (jwforres) [inline app/styles/_container-terminal.less]: not specific to the xterm change, this is an existing bug

File: app/scripts/controllers/pod.js [hunk 3]
```diff
@@ -139,8 +139,8 @@ angular.module('openshiftConsole')
         }
         var windowWidth = win.width();
         var windowHeight = win.height();
-        var termWidth = windowWidth - r.left - 40; // we want 40px right padding, includes 20px padding within the container terminal
-        var termHeight = windowHeight - r.top - 50; // we want 50px bottom padding, includes 20px padding within the container terminal
+        var termWidth = windowWidth - r.left - 54; // we want 40px right padding
+        var termHeight = windowHeight - r.top - 36; // we want 36px bottom padding, includes 6px border within the container terminal
```
Comment (jwforres) [inline app/scripts/controllers/pod.js]: the existing borders actually ended up allowing for really small gutters at certain sizes, this gives better spacing and takes into account the new terminal styles

**Comment by jwforres** [comment]
@spadgett ok PR should be ready to go now

**Comment by jwforres** [comment]
[merge]

**Comment by spadgett** [comment]
```diff
Verifying that checked in built files under dist match the source...
diff --git a/dist/styles/main.css b/dist/styles/main.css
index b84334c..35da42b 100644
--- a/dist/styles/main.css
+++ b/dist/styles/main.css
@@ -5834,4 +5834,4 @@ kubernetes-container-terminal .terminal-actions .spinner{top:5px}
 @media (min-height:700px){.edit-yaml .editor{height:60vh}
 }
 .edit-yaml-errors{font-size:84%;height:2em;text-align:right}
-#from-file .editor{height:350px}
+#from-file .editor{height:350px}
\ No newline at end of file
make: *** [test] Error 1
```

**Comment by spadgett** [comment]
Replaced by #1869

**Pull request closed without merging**